satpy/composites/viirs.py

```diff
@@ -149,11 +149,12 @@ def __call__(self, datasets, optional_datasets, **info):
         return results
 
     def get_angles(self, vis):
+        """Get sun and satellite angles to use in crefl calculations."""
         from pyorbital.astronomy import get_alt_az, sun_zenith_angle
         from pyorbital.orbital import get_observer_look
-
-        lons, lats = vis.attrs['area'].get_lonlats_dask(
-            chunks=vis.data.chunks)
+        lons, lats = vis.attrs['area'].get_lonlats(chunks=vis.data.chunks)
+        lons = da.where(lons >= 1e30, np.nan, lons)
+        lats = da.where(lats >= 1e30, np.nan, lats)
```

Comment:
Should these be a part of `AreaDefinition.get_lonlats` instead ?

Reply:
Good question. I guess it depends how much we want to control what the user gets. Proj from now on will report `inf` values for invalid values so maybe someone has calculations that would prefer those values rather than NaNs? I'm willing to do whatever and don't have strong feelings either way.

Reply:
I'm good with this then.